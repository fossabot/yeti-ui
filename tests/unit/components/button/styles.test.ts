import {
  BLACK,
  PRIMARY,
  WHITE,
} from '../../../../src/styles/colors';
import {
  buttonBase,
  buttonDisable,
  buttonDropdown,
  buttonExpand,
  buttonHollow,
  buttonStyle,
  generateStyle,
} from '../../../../src/components/button/styles';
import {
  getCssValueByProperty,
  resetHeadStyle,
} from '../../../utilities';

describe('src/components/button/styles.ts', () => {
  describe('buttonBase', () => {
    test('should return button base style', () => {
      const baseStyle = buttonBase({});

      expect(baseStyle.fontSize).toBe('0.9rem');
      expect(baseStyle).toMatchSnapshot();
    });

    test('should return button base style with tiny font size', () => {
      const baseStyle = buttonBase({ buttonSize: 'tiny' });

      expect(baseStyle.fontSize).toBe('0.6rem');
    });

    test('should return button base style with small font size', () => {
      const baseStyle = buttonBase({ buttonSize: 'small' });

      expect(baseStyle.fontSize).toBe('0.75rem');
    });

    test('should return button base style with large font size', () => {
      const baseStyle = buttonBase({ buttonSize: 'large' });

      expect(baseStyle.fontSize).toBe('1.25rem');
    });
  });

  describe('buttonDisable', () => {
    test('should return nothing if isDisabled is false', () => {
      const baseStyle = buttonDisable({
        backgroundColor: PRIMARY,
        isDisabled: false,
      });

      expect(baseStyle).toMatchObject({});
    });

    test('should disabled style if isDisabled is true', () => {
      const baseStyle = buttonDisable({
        backgroundColor: PRIMARY,
        isDisabled: true,
      });

      expect(baseStyle).toMatchSnapshot();
    });

    test('should set color to white on a dark background color', () => {
      const baseStyle = buttonDisable({
        backgroundColor: BLACK,
        isDisabled: true,
      });

      expect(baseStyle[':focus'].color).toBe(WHITE);
      expect(baseStyle[':hover'].color).toBe(WHITE);
      expect(baseStyle[':focus'].backgroundColor).toBe(BLACK);
      expect(baseStyle[':hover'].backgroundColor).toBe(BLACK);
    });

    test('should set color to black on a light background color', () => {
      const baseStyle = buttonDisable({
        backgroundColor: WHITE,
        isDisabled: true,
      });

      expect(baseStyle[':focus'].color).toBe(BLACK);
      expect(baseStyle[':hover'].color).toBe(BLACK);
      expect(baseStyle[':focus'].backgroundColor).toBe(WHITE);
      expect(baseStyle[':hover'].backgroundColor).toBe(WHITE);
    });
  });

  describe('buttonDropdown', () => {
    beforeEach(() => {
      resetHeadStyle();
    });

    test('should return nothing if isDropdown is false', () => {
      const baseStyle = buttonDropdown({
        backgroundColor: PRIMARY,
        buttonType: 'primary',
        isDropdown: false,
      });

      expect(baseStyle).toMatchObject({});
    });

    test('should return base dropdown style', () => {
      buttonDropdown({
        backgroundColor: PRIMARY,
        buttonType: 'primary',
        isDropdown: true,
      });

      const display = getCssValueByProperty('display');
      const float = getCssValueByProperty('float');
      const marginLeft = getCssValueByProperty('margin-left');
      const position = getCssValueByProperty('position');
      const top = getCssValueByProperty('top');

      expect(display).toBe('block');
      expect(float).toBe('right');
      expect(marginLeft).toBe('1em');
      expect(position).toBe('relative');
      expect(top).toBe('0.4em');
    });
  });

  describe('buttonExpand', () => {
    test('should return button expand base style if isExpanded is false', () => {
      const baseStyle = buttonExpand({
        isExpanded: false,
      });

      expect(baseStyle).toMatchSnapshot();
    });

    test('should return expand style if isExpanded is true', () => {
      const baseStyle = buttonExpand({
        isExpanded: true,
      });

      expect(baseStyle).toMatchSnapshot();
    });
  });

  describe('buttonHollow', () => {
    test('should return no style if isHollow is false', () => {
      const baseStyle = buttonHollow({
        isHollow: false,
        backgroundColor: PRIMARY,
        backgroundHover: undefined,
        buttonType: 'primary',
      });

      expect(baseStyle).toMatchObject({});
    });

    test('should return no style if isHollow is true', () => {
      const baseStyle = buttonHollow({
        isHollow: false,
        backgroundColor: PRIMARY,
        backgroundHover: undefined,
        buttonType: 'primary',
      });

      expect(baseStyle).toMatchSnapshot();
    });
  });

  describe('buttonStyle', () => {
    test('should return button base style', () => {
      const baseStyle = buttonStyle({
        color: undefined,
        backgroundColor: PRIMARY,
        backgroundHover: undefined,
        buttonType: 'primary',
      });

      expect(baseStyle).toMatchSnapshot();
    });
  });

  // describe('generateStyle', () => {
  //   test('should return button base style', () => {
  //     const baseStyle = buttonBase({});
  //
  //     expect(baseStyle.fontSize).toBe('0.9rem');
  //     expect(baseStyle).toMatchSnapshot();
  //   });
  //
  //   test('should render the button', () => {
  //     expect('this').toMatch('this');
  //   });
  // });
});
