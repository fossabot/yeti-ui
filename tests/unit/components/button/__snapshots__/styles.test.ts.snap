// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/button/styles.ts buttonBase should return button base style 1`] = `
Object {
  "WebkitAppearance": "none",
  "border": "1px solid transparent",
  "borderRadius": "0",
  "cursor": "pointer",
  "display": "inline-block",
  "fontFamily": "inherit",
  "fontSize": "0.9rem",
  "lineHeight": "1",
  "margin": "0 0 1rem 0",
  "padding": "0.85em 1em",
  "textAlign": "center",
  "transition": "background-color 0.25s ease-out, color 0.25s ease-out",
  "verticalAlign": "middle",
}
`;

exports[`src/components/button/styles.ts buttonDisable should disabled style if isDisabled is true 1`] = `
Object {
  ":focus": Object {
    "backgroundColor": "#1779ba",
    "color": "#fefefe",
  },
  ":hover": Object {
    "backgroundColor": "#1779ba",
    "color": "#fefefe",
  },
  "cursor": "not-allowed",
  "opacity": 0.25,
}
`;

exports[`src/components/button/styles.ts buttonExpand should return button expand base style if isExpanded is false 1`] = `
Object {
  "display": "inline-block",
  "margin": "0 0 1rem 0",
  "width": "auto",
}
`;

exports[`src/components/button/styles.ts buttonExpand should return expand style if isExpanded is true 1`] = `
Object {
  "display": "block",
  "marginLeft": "0",
  "marginRight": "0",
  "width": "100%",
}
`;

exports[`src/components/button/styles.ts buttonHollow should return no style if isHollow is true 1`] = `Object {}`;

exports[`src/components/button/styles.ts buttonStyle should return button base style 1`] = `
Object {
  ":focus": Object {
    "backgroundColor": "#126195",
    "color": "#fefefe",
  },
  ":hover": Object {
    "backgroundColor": "#126195",
    "color": "#fefefe",
  },
  "backgroundColor": "#1779ba",
  "color": "#fefefe",
  "textDecoration": "none",
}
`;
